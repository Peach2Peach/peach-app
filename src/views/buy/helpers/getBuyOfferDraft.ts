type GetBuyOfferDraftParams = {
  minBuyAmount: number
  maxBuyAmount: number
  meansOfPayment?: MeansOfPayment
}
export const getBuyOfferDraft = ({
  minBuyAmount,
  maxBuyAmount,
  meansOfPayment = {},
}: GetBuyOfferDraftParams): BuyOfferDraft => ({
  type: 'bid',
  creationDate: new Date(),
  lastModified: new Date(),
  amount: [minBuyAmount, maxBuyAmount],
  meansOfPayment,
  paymentData: {},
  releaseAddress: '',
  originalPaymentData: [],
})
