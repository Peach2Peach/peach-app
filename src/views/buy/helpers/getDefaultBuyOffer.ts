type DefaultOfferProps = {
  minBuyAmount: number
  maxBuyAmount: number
  meansOfPayment?: MeansOfPayment
  kyc?: boolean
}
export const getDefaultBuyOffer = ({
  minBuyAmount,
  maxBuyAmount,
  meansOfPayment = {},
  kyc = false,
}: DefaultOfferProps): BuyOfferDraft => ({
  type: 'bid',
  creationDate: new Date(),
  lastModified: new Date(),
  amount: [minBuyAmount, maxBuyAmount],
  meansOfPayment,
  paymentData: {},
  releaseAddress: '',
  originalPaymentData: [],
  kyc,
})
