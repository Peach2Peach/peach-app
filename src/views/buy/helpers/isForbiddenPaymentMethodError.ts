import { isPaymentMethod } from '../../../utils/validation/isPaymentMethod'

export const isForbiddenPaymentMethodError = (
  errorMessage: string | null,
  errorDetails: unknown,
): errorDetails is PaymentMethod[] =>
  errorMessage === 'FORBIDDEN' && Array.isArray(errorDetails) && errorDetails.every(isPaymentMethod)
