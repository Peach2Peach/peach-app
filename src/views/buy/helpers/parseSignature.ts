const signaturePattern = /[a-zA-Z0-9/=+]{88}/u

export const parseSignature = (signature: string) => signature.match(signaturePattern)?.pop() || signature
