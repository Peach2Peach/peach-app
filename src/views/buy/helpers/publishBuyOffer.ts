import { publishPGPPublicKey } from '../../../init/publishPGPPublicKey'
import { saveOffer } from '../../../utils/offer'
import { postBuyOffer } from '../../../utils/peachAPI'

export const publishBuyOffer = async (offerDraft: BuyOfferDraft) => {
  let [result, err] = await postBuyOffer(offerDraft)

  if (err?.error === 'PGP_MISSING') {
    await publishPGPPublicKey()
    const response = await postBuyOffer(offerDraft)
    result = response[0]
    err = response[1]
  }
  if (result) {
    saveOffer({ ...offerDraft, ...result })
    return { offerId: result.id, isOfferPublished: true, errorMessage: null }
  }
  return { isOfferPublished: false, errorMessage: err?.error || 'POST_OFFER_ERROR' }
}
