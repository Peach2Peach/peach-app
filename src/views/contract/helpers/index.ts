export { isPSBTForBatch } from "./isPSBTForBatch";
export { shouldShowTradeStatusInfo } from "./shouldShowTradeStatusInfo";
export { verifyReleasePSBT as verifyPSBT } from "./verifyReleasePSBT";
