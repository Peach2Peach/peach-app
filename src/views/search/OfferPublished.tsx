import { ReactElement } from 'react'
import { View } from 'react-native'
import { Icon, PrimaryButton, Text } from '../../components'
import tw from '../../styles/tailwind'
import i18n from '../../utils/i18n'
import { useOfferPublishedSetup } from './hooks/useOfferPublishedSetup'

export default (): ReactElement => {
  const { goToOffer, shouldGoBack, closeAction } = useOfferPublishedSetup()

  return (
    <View style={tw`items-center justify-between h-full px-9 pb-7`}>
      <View style={tw`justify-center flex-shrink w-full h-full`}>
        <Text style={tw`text-center h4 text-primary-background-light`}>{i18n('offer.published.title')}</Text>
        <View style={tw`flex-row items-center mt-8 gap-6`}>
          <Icon id="checkCircleInverted" style={tw`w-16 h-16`} color={tw`text-primary-background-light`.color} />
          <Text style={tw`flex-shrink body-l text-primary-background-light`}>{i18n('offer.published.description')}</Text>
        </View>
      </View>
      <PrimaryButton white narrow onPress={goToOffer}>
        {i18n('showOffer')}
      </PrimaryButton>
      <PrimaryButton style={tw`mt-3`} white narrow border onPress={closeAction}>
        {i18n(shouldGoBack ? 'close' : 'home')}
      </PrimaryButton>
    </View>
  )
}
