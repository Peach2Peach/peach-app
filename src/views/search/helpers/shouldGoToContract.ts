export const shouldGoToContract = (error: APIError): error is APIError & { details: { contractId: string } } =>
  !!error.details
  && typeof error.details === 'object'
  && 'contractId' in error.details
  && typeof error.details.contractId === 'string'
