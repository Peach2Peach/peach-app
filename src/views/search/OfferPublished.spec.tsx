import { fireEvent, render } from '@testing-library/react-native'
import { NavigationWrapper, goBackMock, replaceMock } from '../../../tests/unit/helpers/NavigationWrapper'
import OfferPublished from './OfferPublished'

const offerId = '123'
const useRouteMock = jest.fn().mockReturnValue({ params: { offerId, isSellOffer: false, shouldGoBack: false } })
jest.mock('../../hooks/useRoute', () => ({
  useRoute: () => useRouteMock(),
}))

describe('OfferPublished', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<OfferPublished />, { wrapper: NavigationWrapper })
    expect(toJSON()).toMatchSnapshot()
  })

  it('should go to search when pressing show offer button', () => {
    const { getByText } = render(<OfferPublished />, { wrapper: NavigationWrapper })
    const button = getByText('show offer')
    fireEvent.press(button)
    expect(replaceMock).toHaveBeenCalledWith('search', { offerId })
  })
  it('should go back to buy for buy offers when pressing back close button', () => {
    const { getByText } = render(<OfferPublished />, { wrapper: NavigationWrapper })
    const closeButton = getByText('close')
    fireEvent.press(closeButton)
    expect(replaceMock).toHaveBeenCalledWith('buy')
  })
  it('should go back to sell for sell offers when pressing back close button', () => {
    useRouteMock.mockReturnValueOnce({ params: { isSellOffer: true, shouldGoBack: false } })
    const { getByText } = render(<OfferPublished />, { wrapper: NavigationWrapper })
    const closeButton = getByText('close')
    fireEvent.press(closeButton)
    expect(replaceMock).toHaveBeenCalledWith('sell')
  })
  it('should go back when pressing close', () => {
    useRouteMock.mockReturnValueOnce({ params: { isSellOffer: true, shouldGoBack: true } })
    const { getByText } = render(<OfferPublished />, { wrapper: NavigationWrapper })
    const closeButton = getByText('close')
    fireEvent.press(closeButton)
    expect(goBackMock).toHaveBeenCalled()
  })
})
