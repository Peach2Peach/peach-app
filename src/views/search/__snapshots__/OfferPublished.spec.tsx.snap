// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferPublished renders correctly 1`] = `
[
  <View
    style={
      {
        "flexGrow": 1,
        "gap": 32,
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        [
          {
            "color": "#2B1911",
            "fontFamily": "Baloo2-Regular",
            "fontSize": 16,
            "letterSpacing": 0.16,
            "lineHeight": 24,
          },
          {
            "color": "#FFFCFA",
            "flexShrink": 1,
            "fontFamily": "Baloo2-Bold",
            "fontSize": 34,
            "letterSpacing": 0.34,
            "lineHeight": 42,
            "textAlign": "center",
          },
          false,
        ]
      }
    >
      offer published!
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 24,
        }
      }
    >
      <test-file-stub
        fill="#FFFCFA"
        style={
          [
            {
              "height": 92,
              "width": 92,
            },
            undefined,
          ]
        }
      />
      <Text
        allowFontScaling={false}
        style={
          [
            {
              "color": "#2B1911",
              "fontFamily": "Baloo2-Regular",
              "fontSize": 16,
              "letterSpacing": 0.16,
              "lineHeight": 24,
            },
            [
              {
                "color": "#FFFCFA",
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Baloo2-Regular",
                "fontSize": 20,
                "letterSpacing": 0.2,
                "lineHeight": 28,
              },
              false,
            ],
            false,
          ]
        }
      >
        We'll send you a notification when there are matchesÂ available.
      </Text>
    </View>
  </View>,
  <View
    style={
      {
        "alignSelf": "center",
        "gap": 12,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFCFA",
          "borderColor": undefined,
          "borderRadius": 9999,
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
          "minWidth": 104,
          "opacity": 1,
          "paddingBottom": 3,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 3,
        }
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          [
            {
              "color": "#2B1911",
              "fontFamily": "Baloo2-Regular",
              "fontSize": 16,
              "letterSpacing": 0.16,
              "lineHeight": 24,
            },
            [
              {
                "fontFamily": "Baloo2-SemiBold",
                "fontSize": 13,
                "letterSpacing": 0.47,
                "lineHeight": 22,
                "textTransform": "uppercase",
              },
              {},
              {
                "color": "#F56522",
              },
            ],
            false,
          ]
        }
      >
        show offer
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#FFFCFA",
          "borderRadius": 9999,
          "borderWidth": 2,
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
          "minWidth": 104,
          "opacity": 1,
          "paddingBottom": 3,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 3,
        }
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          [
            {
              "color": "#2B1911",
              "fontFamily": "Baloo2-Regular",
              "fontSize": 16,
              "letterSpacing": 0.16,
              "lineHeight": 24,
            },
            [
              {
                "fontFamily": "Baloo2-SemiBold",
                "fontSize": 13,
                "letterSpacing": 0.47,
                "lineHeight": 22,
                "textTransform": "uppercase",
              },
              {},
              {
                "color": "#FFFCFA",
              },
            ],
            false,
          ]
        }
      >
        close
      </Text>
    </View>
  </View>,
]
`;
